### ISSUE TITLE
Outline Breadcrumb Navigation Pattern

### ISSUE THREAD

**Issue by michael-collins** [elmsln/elmsln#162]
Implement the navigation pattern in the link for book outline nodes. This will undergo iterative improvement throughout the year. https://files.slack.com/files-pri/T02F8NY7Y-F08F269PA/breadcrumbnav.jpg

**Comment by btopro** [comment]
I see a lot of assignment and unassignment but not code

**Comment by michael-collins** [comment]
I didn't realize that only one person could be assigned at a time. Also, 1 hour to find internet, 1.5 hours of npm not working. Will try again.

**Comment by btopro** [comment]
does a static html mockup require npm now?

**Comment by michael-collins** [comment]
Only if you want to use SASS.

**Comment by btopro** [comment]
![2015-08-07_00-40-53](https://cloud.githubusercontent.com/assets/329735/9128856/f82702d2-3c9c-11e5-8a63-240b431cf3df.png)

current state after today. live demo -- https://drupal.psu.edu/library/

**Comment by aitala** [comment]
Massively picky comment..  child menu  ‘opens’ towards the left, parent 'opens' to the right.  I know that it is within some sort of box/div..  just feels odd..

e

![screen shot 2015-08-07 at 9 35 13 am](https://cloud.githubusercontent.com/assets/4474470/9136898/13448526-3ce8-11e5-9d0f-a3152c41562e.png)

**Comment by juststormy** [comment]
My comment is the ellipses should be to the right of the dropdown arrow " v... " Maybe that is just my editor background... ellipses mean "more" later... not before...

**Comment by btopro** [comment]
![2015-08-10_06-50-36](https://cloud.githubusercontent.com/assets/329735/9169909/584c1404-3f2c-11e5-9e37-3e18d9aef8f2.png)

Feedback from latest modifications. When on a page that has children pages it should still highlight in some fashion to indicate that the page is active. This happens on the lowest page levels.

adding a commit that present an example step in rem's for supporting REALLY deep structures/ While its ridiculous to have 9 children it is technically possible via drupal's menu / book structure which maxes out at 9 parents / children

**Comment by btopro** [comment]
From my notes for some more things to do:
- [ ] Query cache menu calls in book cache somehow
- [ ] Make cache command wrappers for all my parents and all my sibling type of calls
- [ ] Also cache in a way we can target to expire easily
- [ ] Call for node access during rendering of items (right now it shows everything / doesn't support hidden nodes even though clicking to it will still give access denied)
- [ ] Make the left nav stop showing the black bar / auto label everything if available
- [ ] Allow left nav to stop showing a label (option to do so)
- [ ] Bring nav into alignment with the breadcrumbs
- [ ] Provide support for multiple levels of auto labeling, not just highest, same for auto numbering
- [ ] Option to hide title on page in settings

**Comment by heyMP** [comment]
@btopro how big of an issue would it be to move the book navigation into the navigation region `nav.etb-tool`?

**Comment by btopro** [comment]
Oh, that's what I meant by "Bring nav into alignment with the breadcrumbs". Issue as far as wanting it, def want it. Issue as far as doing it, well, that might take a little digging. Right now that'd delivered by the page tpl and this is being rendered as a block. I think we'd need to make a region for block placement, either repurposing one that's not used now or creating a new one that's right in the middle of that bar which could provide a nice area for breadcrumb nav when in other systems.

I'd vote for creating a new one (even though it's annoying) and then updating all the sub-theme files to reflect having this new area. Then the context that currently places the book nav block in the header could just place it in this new breadcrumb area and we're g2g.

**Comment by btopro** [comment]
#190 related

**Comment by btopro** [comment]
splitting "Provide support for multiple levels of auto labeling, not just highest, same for auto numbering" off to a new issue; the others are marked resolved as we are killing the nav #201
